pkg/flare/flare.go

```diff
@@ -0,0 +1,403 @@
+	"github.com/octarinesec/secret-detector/pkg/secrets"
+	"gopkg.in/yaml.v3"
+)
+
+// defaults
+const (
+	defaultSecretReplacementValue = "<REDACTED>"
+)
+
+var defaultEnvVarPrefixes = []string{"LAKEFS_", "HTTP_", "HOSTNAME"}
+
+const (
+	DirPermissions  = 0700
+	FilePremissions = 0600
+	FlareUmask      = 002
+)
+
+// LogFormat is a log file format supported by the flare package
+type LogFormat string
+
+const (
+	LogFormatJSON      LogFormat = "json"
+	LogFormatPlainText LogFormat = "text"
+)
+
+var (
+	ErrExtractDateFromJSONLogLine = errors.New("failed to extract date from log line")
+	ErrDateNotFound               = errors.New("date not found in log line")
+
+	secretScanner        secrets.Scanner
+	secretScannerInitErr error
+
+	plainTextLogDateRegex = regexp.MustCompile(`\[\d{4}(.\d{2}){2}(\s|T)(\d{2}.){2}\d{2}[\+-]?\d{2}:\d{2}\]`)
+)
+
+type WithTime struct {
+	Time time.Time
+}
+
+type Flare struct {
+	envVarPrefixes         []string
+	secretReplacementValue string
+	logFormat              LogFormat
+	// LogDateLayout is the layout used by time.Parse to parse dates in log lines.
+	// The default value is time.RFC3339.
+	// This can be changed using the WithLogDateLayout option.
+	LogDateLayout string
+}
+
+type Option func(*Flare)
+
+func NewFlare(logFormat LogFormat, options ...Option) (*Flare, error) {
+	if secretScannerInitErr != nil {
+		return nil, fmt.Errorf("failed to init secrets scanner: %w", secretScannerInitErr)
+	}
+	flare := &Flare{
+		envVarPrefixes:         defaultEnvVarPrefixes,
+		secretReplacementValue: defaultSecretReplacementValue,
+		LogDateLayout:          time.RFC3339,
+		logFormat:              logFormat,
+	}
+
+	for _, opt := range options {
+		opt(flare)
+	}
+
+	return flare, nil
+}
+
+// WithEnvVarPrefixes replaces the default list of environment variable prefixes that flare processes.
+// The default list is "LAKEFS_", "HTTP_", "HOSTNAME".
+func WithEnvVarPrefixes(prefixes []string) Option {
+	return func(f *Flare) {
+		f.envVarPrefixes = prefixes
+	}
+}
+
+// WithAdditionalEnvVarPrefix adds additional environment prefixes to the default list that flare processes.
+// The default list is "LAKEFS_", "HTTP_", "HOSTNAME".
+func WithAdditionalEnvVarPrefix(envVar string) Option {
+	return func(f *Flare) {
+		f.envVarPrefixes = append(f.envVarPrefixes, envVar)
+	}
+}
+
+// WithSecretReplacementValue replaces the default secret replacement value of "<REDACTED>".
+func WithSecretReplacementValue(v string) Option {
+	return func(f *Flare) {
+		f.secretReplacementValue = v
+	}
+}
+
+// WithLogDateLayout sets an alternate date layout for parsing dates in log lines.
+func WithLogDateLayout(l string) Option {
+	return func(f *Flare) {
+		f.LogDateLayout = l
+	}
+}
+
+// LogLineDateExtractor represents a function that extracts the date from a log line in a specific format.
+type LogLineDateExtractor func(line, logDateLayout string) (time.Time, error)
+
+var dateExtractors = map[LogFormat]LogLineDateExtractor{
```

Comment:
I'm not sure about the date range relevant. 
1. It's really not ergonomic for a user, it took me like 10 minutes to get the ISO timestsamp right 
2. it added complexity to the code. 

Why is it here? 

Suggestion: if too many logs is a concern add regex filter for file names that can exclude / include file by names.

Reply:
Log filter flags can now be provided in any known Go `time.Time` layout.